File: Ruby/TP8/Ej4&9/ellipse.rb
require_relative 'figure'
class Ellipse < Figure
  def initialize(center, mayor_dim, minor_dim)
    @center= center
    @mayor_dim = mayor_dim
    @minor_dim = minor_dim
    @rad1= @mayor_dim/2.0
    @rad2=@minor_dim/2.0
  end
  def to_s
    'Ellipse [Centro: '+@center.to_s+
        ', DMayor:'+@mayor_dim.to_s +
        ', Dmenor:'+@minor_dim.to_s+']'
  end
  def perimeter
    Math::PI*2*Math.sqrt((@rad1**2+@rad2**2)/2)
  end
  def area
    Math::PI*@rad1*@rad2
  end
end

File: Extras/Parciales/Figures/Ellipse.java
package ar.itba.edu.Parciales.Figures;

public class Ellipse extends Figure {
    protected final Point centerPoint;
    protected final double sMayorAxis, sMinorAxis;

    public Ellipse(Point centerPoint, double sMayorAxis, double sMinorAxis) {
        this.centerPoint = centerPoint;
        this.sMayorAxis = sMayorAxis;
        this.sMinorAxis = sMinorAxis;
    }

    @Override
    double area() {
        return Math.PI/4*sMayorAxis*sMinorAxis;
    }

    @Override
    double perimeter() {
        return Math.PI*(sMinorAxis+sMayorAxis);
    }

    @Override
    public String toString() {
        return String.format("Ellipse[Centro %s, DMayor %.2f, DMenor %.2f]",centerPoint,sMayorAxis,sMinorAxis);
    }
}
